reference/reference.go

```diff
@@ -195,40 +204,139 @@ func ParseNamed(s string) (Named, error) {
 	return named, nil
 }
 
-// WithName returns a named object representing the given string. If the input
-// is invalid ErrReferenceInvalidFormat will be returned.
-func WithName(name string) (Named, error) {
```

Comment:
Is this not identical to `ParseNamed`?

It seems there are two paths here: one path uses `ParseNamed` + `NamedOnly`/`IsNameOnly` and the other uses this function (or what would be `ParseNameOnly`). Do we need to ensure that we only use the repository name or do need to ensure that the input is strictly the repository name?